modeltranslation/fields.py

```diff
@@ -156,8 +158,53 @@ def __init__(self, translated_field, language, empty_value, *args, **kwargs):
         # (will show up e.g. in the admin).
         self.verbose_name = build_localized_verbose_name(translated_field.verbose_name, language)
 
+        # M2M support - <rewrite related_name> <patch intermediary model>
+        if isinstance(self.translated_field, fields.related.ManyToManyField) and hasattr(
+            self.remote_field, "through"
+        ):
+            import copy
+
+            # Since fields cannot share the same remote_field object:
+            self.remote_field = copy.copy(self.remote_field)
+
+            # To support multiple relations to self, must provide a non null language scoped related_name
+            if self.remote_field.symmetrical and (
+                self.remote_field.model == "self"
+                or self.remote_field.model == self.model._meta.object_name
+                or self.remote_field.model == self.model
+            ):
+                self.remote_field.related_name = "%s_rel_+" % self.name
+            elif self.remote_field.is_hidden():
+                # Even if the backwards relation is disabled, django internally uses it, need to use a language scoped related_name
+                self.remote_field.related_name = "_%s_%s_+" % (
+                    self.model.__name__.lower(),
+                    self.name,
+                )
+            else:
+                # Default case with standard related_name must also include language scope
+                if self.remote_field.related_name is None:
+                    # For implicit related_name use different query field name
+                    loc_related_query_name = build_localized_fieldname(
+                        self.related_query_name(), self.language
+                    )
+                    self.related_query_name = lambda: loc_related_query_name
+                    self.remote_field.related_name = "%s_set" % (
+                        build_localized_fieldname(self.model.__name__.lower(), language),
+                    )
+                else:
+                    self.remote_field.related_name = build_localized_fieldname(
+                        self.remote_field.get_accessor_name(), language
+                    )
+
+            # Patch intermediary model with language scope to create correct db table
+            self.remote_field.through = patch_intermediary_model(self, language, translator)
+            self.remote_field.field = self  # Django 1.6
+
+            if hasattr(self.remote_field.model._meta, '_related_objects_cache'):
+                del self.remote_field.model._meta._related_objects_cache
+
         # ForeignKey support - rewrite related_name
-        if not NEW_RELATED_API and self.rel and self.related and not self.rel.is_hidden():
+        elif not NEW_RELATED_API and self.rel and self.related and not self.rel.is_hidden():
             import copy
```

Comment:
Oh, probably that's why you imported `copy` locally. Let's move this to top-level also.

Reply:
That's the reason I'll move everything. I tried to mimic as much as possible the code I've found.